src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs

```diff
@@ -61,30 +63,38 @@ public class CryptoCompareWebsocketHandler : ClientWebsocketRedirectHandlerBase<
 
     protected override async Task<bool> AddInternalAsync(TopicSubscription subscription)
     {
-        var jDoc = JsonDocument.Parse(subscription.Topic);
-        JsonProperty? unsubProperties = jDoc.RootElement.EnumerateObject().Any(t => t.Name.Equals("action", StringComparison.InvariantCultureIgnoreCase)
-            && t.Value.GetString().Equals("SubRemove", StringComparison.InvariantCultureIgnoreCase)) ?
-            jDoc.RootElement.EnumerateObject().FirstOrDefault(t => t.Name == "subs")
-            : null;
-        if (unsubProperties != null)
+        if (HasRemoveAction(subscription))
         {
-            var unsubArrayLength = unsubProperties.Value.Value.GetArrayLength();
-            for (int i = 0; i < unsubArrayLength; i++)
+            return await RemoveAsync(subscription);
+        }
+        return await base.AddInternalAsync(subscription);
+    }
+
+    public async Task<bool> RemoveAsync(TopicSubscription subscription)
+    {
+        var payload = JsonSerializer.Deserialize<CryptoCompareSubscription>(subscription.Topic);
+
+        if (payload == null) return false;
+        if (payload.Action != SubscribeActions.SubRemove) return false;
+
+        foreach (var item in payload.Subs)
+        {
+            var toRemoveSubs = Subscriptions.Where(t => t.Topic.ContainsIgnoreCase(item)).ToList();
+            foreach (var unwanted in toRemoveSubs)
             {
-                var unsubElement = unsubProperties.Value.Value[i].GetString();
-                var toRemoveSubs = Subscriptions.Where(t => t.Topic.Contains(unsubElement, StringComparison.InvariantCultureIgnoreCase))
-                    .ToList();
-                if (toRemoveSubs.Any())
-                {
-                    await this.RemoveAsync(toRemoveSubs);
-                }
+                Subscriptions.Remove(unwanted);
             }
-            SendClient(subscription.Topic);
-            return true;
         }
 
-        return await base.AddInternalAsync(subscription);
+        SendClient(subscription.Topic);
+
+        await Task.CompletedTask;
+        return true;
     }
 
-    protected override Uri ClientSocketUri => _config.WebSocketEndpoint;
+    private static bool HasRemoveAction(TopicSubscription subscription)
+    {
+        var remove = SubscribeActions.SubRemove.ToString();
```

Comment:
Can you add some summary comments to methods and maybe choose a different name for the variable 'remove' I have difficulty to understand what this method does

Reply:
Sure, done.